src/util.c

```diff
@@ -573,7 +573,7 @@ cc_oci_enable_networking (void)
  * \return \c guint32 in big endian order.
  */
 guint32 cc_oci_get_big_endian_32(const guint8 *buf) {
-	return (guint32)(buf[0] >> 24 | buf[1] >> 16 | buf[2] >> 8 | buf[3]);
+	return (guint32)(buf[0] << 24 | buf[1] << 16 | buf[2] << 8 | buf[3]);
```

Comment:
This demonstrates that we really need some unit tests for these functions. Could you look into that on a separate PR?